lib/iris/fileformats/netcdf.py

```diff
@@ -724,7 +724,23 @@ def write(self, cube, local_keys=None, unlimited_dimensions=None,
             `chunksizes` and `endian` keywords are silently ignored for netCDF
             3 files that do not use HDF5.
 
+        .. deprecated:: 1.8.0
+
+            NetCDF default saving behaviour currently assigns the outermost
+            dimensions to unlimited. This behaviour is to be deprecated, in
+            favour of no automatic assignment. To switch to the new behaviour,
+            set `iris.FUTURE.netcdf_no_unlimited` to True.
+
         """
+        msg = ('NetCDF default saving behaviour currently assigns the '
+               'outermost dimensions to unlimited. This behaviour is to be '
+               'deprecated, in favour of no automatic assignment. To switch '
+               'to the new behaviour, set iris.FUTURE.netcdf_no_unlimited to '
+               'True.')
+        if (unlimited_dimensions is None and not
+                iris.FUTURE.netcdf_no_unlimited):
+            warnings.warn(msg)
```

Comment:
This is fine :+1: ... but it also needs to be defined on the top-level `save` function in this module.